### ISSUE TITLE
JVMArguments passed to application, rather than JVM

### ISSUE THREAD

**Issue by liamsharp** [tofi86/universalJavaApplicationStub#46]
In my fork of the universalJavaApplicationStub I noticed that the JVMArguments are actually passed to the Java app being run, rather than the JVM itself. So I just switched 

```
${JVMMainClass}\
${JVMArguments:+ $JVMArguments}\
${ArgsPassthru:+ $ArgsPassthru}
```

to:

```
${JVMArguments:+ $JVMArguments}\
${JVMMainClass}\
${ArgsPassthru:+ $ArgsPassthru}
```

I'm using https://github.com/tofi86/universalJavaApplicationStub/blob/f9011ddd69b7c8afb0230c5942a6944c3d69b76b/src/universalJavaApplicationStub as a start point.

**Comment by tofi86** [comment]
Thanks for reporting this. Looks indeed like a bug, I will take a closer look the next week.

**Comment by tofi86** [comment]
Sorry for the late response.

After digging deeper, I now mark this issue as `invalid` because the code works correct.
The variable `$JVMArguments` you mentioned is probably inconsistently named and I changed that with commit xxxxxxxx

Let me explain:

This is a snippet from an Apple style Plist file:

```xml
<key>JavaX</key>
<dict>
  <key>MainClass</key>
  <string>org.main.class</string>
  <key>JVMVersion</key>
  <string>1.8</string>
  <key>ClassPath</key>
  <array>
    <string>$JAVAROOT/package.jar</string>
  </array>
  <key>SplashFile</key>
  <string>SplashScreen_Mac.png</string>
  <key>VMOptions</key>
  <string>-Xmx1024m -Xms512m</string>
  <key>Arguments</key>
  <string>-a b -c=d</string>
  <key>Properties</key>
  <dict>
    <key>apple.laf.useScreenMenuBar</key>
    <string>true</string>
  </dict>
</dict>
```

And this is the function which sets the `$JVMArguments` variable from the `Arguments` Plist Key:

```bash
# read the JVM Arguments
JVMArguments=`/usr/libexec/PlistBuddy -c "print ${JavaKey}:Arguments" "${InfoPlistFile}" 2> /dev/null | xargs`
# replace occurences of $APP_ROOT with its content
JVMArguments=`eval "echo ${JVMArguments}"`
```

However, when looking at the Oracle style Plist file:

```xml
<key>JVMMainClassName</key>
<string>org.main.class</string>
<key>JVMOptions</key>
<array>
  <string>-Dapple.laf.useScreenMenuBar=true</string>
  <string>-Dcom.apple.mrj.application.apple.menu.about.name=Hello</string>
</array>
<key>JVMDefaultOptions</key>
<dict>
  <key>Xmx</key>
  <string>-Xms512M</string>
  <key>Xms</key>
  <string>-Xmx1024M</string>
</dict>
<key>JVMArguments</key>
<array>
  <string>-a b</string>
  <string>-c=d</string>
</array>
```
evaluated by

```bash
# read the JVM Arguments
JVMArguments=`/usr/libexec/PlistBuddy -c "print :JVMArguments" "${InfoPlistFile}" 2> /dev/null | tr -d '\n' | sed -E 's/Array \{ *(.*) *\}/\1/g' | sed 's/  */ /g' | xargs`
# replace occurences of $APP_ROOT with its content
JVMArguments=`eval "echo ${JVMArguments}"`
```

we see, that Oracle once named the same key `JVMArguments` which is just not correct.
Unfortunately the docs for the Oracle style Plist do not exist anymore, but it's also how [infinitekind's fork](https://bitbucket.org/infinitekind/appbundler) is [handling (1)](https://bitbucket.org/infinitekind/appbundler/src/97eb6c13cf33e36a9e1efdb8f08de59cdfc12772/appbundler/src/com/oracle/appbundler/AppBundlerTask.java?at=default&fileviewer=file-view-default#AppBundlerTask.java-732) [it (2)](https://bitbucket.org/infinitekind/appbundler/src/97eb6c13cf33e36a9e1efdb8f08de59cdfc12772/appbundler/native/main.m?at=default&fileviewer=file-view-default#main.m-342).

I renamed my local bash variable from `$JVMArguments` to `$MainArgs` to make this clear.

If you experienced wrong JVM arguments being passed to the main class, you may have used the wrong Plist key to set the JVM options.
Can you paste a snippet of your Plist file?

**Comment by tofi86** [comment]
I also added a table with supported PList keys and their function to the README file with commit 1e9c28710d984484d2578693a5f3895ec64d8ba9

**Comment by tofi86** [comment]
Closing as invalid.
Feel free to reopen if necessary and if you think I mixed something up...

**Comment by tofi86** [comment]
The updated README file with PList key description has been released with [version 3.0.0](https://github.com/tofi86/universalJavaApplicationStub/releases/tag/v3.0.0) right now...
